crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -87,24 +87,10 @@ impl<F: Send + Clone> GetTracker<F, PaymentData<F>, api::PaymentsRequest> for Pa
                 error.to_not_found_response(errors::ApiErrorResponse::PaymentNotFound)
             })?;
         payment_attempt.payment_method = payment_method_type.or(payment_attempt.payment_method);
-
-        payment_attempt.payment_method = payment_method_type.or(payment_attempt.payment_method);
         payment_attempt.browser_info = browser_info;
         currency = payment_attempt.currency.get_required_value("currency")?;
         amount = payment_attempt.amount.into();
 
-        connector_response = db
-            .find_connector_response_by_payment_id_merchant_id_attempt_id(
-                &payment_attempt.payment_id,
-                &payment_attempt.merchant_id,
-                &payment_attempt.attempt_id,
-                storage_scheme,
-            )
-            .await
-            .map_err(|error| {
-                error.to_not_found_response(errors::ApiErrorResponse::PaymentNotFound)
-            })?;
-
         let shipping_address = helpers::get_address_for_payment_request(
```

Comment:
We need to verify that the customer_id exists before checking for saved addresses.

Reply:
Also the function `get_address_for_payment_request` needs to accept a customer_id instead of Option<customer_id>